task_DEV_4/task_DEV_4/Parallelogram.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+namespace task_DEV_4
+{
+    class Parallelogram : Square
+    {
+        public double Height { get; set; }
+        public double Angle { get; set; }
+        public double Side2 { get; set; }
+
+        public Parallelogram(double side, double height) : base(side)
+        {
+            Height = height;
+        }
+
+        public Parallelogram(double side, double side2, double angle) : base(side)
+        {
+            Side2 = side2;
+            Angle = angle;
+        }
+
+        public override double CalculateArea()
+        {
+            if (Height > 0 && Side > 0)
+            {
+                return Side * Height;
+            }
+            else
```

Comment:
There should be 
else if (Side > 0 && Side2 >0 && Angle > 0)
{
    return Side * Side2 * Math.Sin(Angle);
}
else
{
    throw new SomeException("custom message")
}

Reply:
Because we can't be sure that Side, Side2, Angle will be set.

Reply:
Done